src/zope/app/container/browser/contents.py

```diff
@@ -204,6 +205,11 @@ def renameObjects(self):
         ids = request.get("rename_ids")
         newids = request.get("new_value")
 
+        if isinstance(ids, six.string_types):
+            ids = [ids]
+        if isinstance(newids, six.string_types):
+            newids = [newids]
```

Comment:
I believe it has to do with the change from `zope.app.testing` to `zope.app.wsgi`. The `z.a.testing` method directly creates a request object, filling in the parameters directly from the objects passed, and then handing that request off to `publish`. `z.a.wsgi`, on the other hand, converts the objects into a form-encoded body that then gets treated like bytes-on-the-wire and parsed. 

So here the test code was passing a list, and `z.a.testing` made sure that the list always got through. But once `z.a.wsgi` is involved, lists of one item (and only lists of one item, IIRC) started arriving as just a single string, and not a list. Lists of multiple items still came in as lists (again, IIRC). 

I don't know if that's expected or a failure somewhere.

The encoding issue looks to be the same thing. The test code is directly passing things that cannot be round-tripped (through UTF-8?) in the way that the `z.a.wsgi` code wants to.